src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.cs

```diff
@@ -0,0 +1,392 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace Internal.NativeCrypto
+{
+    internal static partial class CapiHelper
+    {
+        // Provider type to use by default for DSS operations.
+        internal const int DefaultDssProviderType = (int)ProviderType.PROV_DSS_DH;
+
+        internal const int DSS_MAGIC = 0x31535344;          // Encoding of "DSS1"
+        internal const int DSS_PRIVATE_MAGIC = 0x32535344;  // Encoding of "DSS2"
+        internal const int DSS_PUB_MAGIC_VER3 = 0x33535344; // Encoding of "DSS3"
+        internal const int DSS_PRIV_MAGIC_VER3 = 0x34535344;// Encoding of "DSS4"
+
+        private const int DSS_Q_LEN = 20;
+
+        /// <summary>
+        /// Check to see if a better CSP than the one requested is available
+        /// DSS providers are supersets of each other in the following order:
+        ///    1. MS_ENH_DSS_DH_PROV
+        ///    2. MS_DEF_DSS_DH_PROV
+        ///
+        /// This will return the best provider which is a superset of wszProvider,
+        /// or NULL if there is no upgrade available on the machine.
+        /// </summary>
+        /// <param name="dwProvType">provider type</param>
+        /// <param name="wszProvider">provider name</param>
+        /// <returns>Returns upgrade CSP name</returns>
+        public static string UpgradeDSS(int dwProvType, string wszProvider)
+        {
+            string wszUpgrade = null;
+            if (string.Equals(wszProvider, MS_DEF_DSS_DH_PROV, StringComparison.Ordinal))
+            {
+                SafeProvHandle safeProvHandle;
+                // If this is the base DSS/DH provider, see if we can use the enhanced provider instead.
+                if (S_OK == AcquireCryptContext(out safeProvHandle,
+                    null,
+                    MS_ENH_DSS_DH_PROV,
+                    dwProvType,
+                    (uint)CryptAcquireContextFlags.CRYPT_VERIFYCONTEXT))
+                {
+                    wszUpgrade = MS_ENH_DSS_DH_PROV;
+                }
+                safeProvHandle.Dispose();
+            }
+            return wszUpgrade;
+        }
+
+        /// <summary>
+        /// Helper for DsaCryptoServiceProvider.ImportParameters()
+        /// </summary>
+        internal static byte[] ToKeyBlob(this DSAParameters dsaParameters, int algId)
+        {
+            if (algId != CapiHelper.CALG_DSS_SIGN)
+                throw new PlatformNotSupportedException();
```

Comment:
Removed it. The pattern was borrowed from RSA which does the same thing with the comment:
            // The original FCall this helper emulates supports other algId's - however, the only algid we need to support is CALG_RSA_KEYX. We will not
            // port the codepaths dealing with other algid's.